src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -0,0 +1,132 @@
+import { Component } from '../Base/Component';
+import { l } from '../../strings/Strings';
+import { IFieldOption, ComponentOptions } from '../Base/ComponentOptions';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { $$, Dom } from '../../utils/Dom';
+import { Initialization } from '../Base/Initialization';
+import { exportGlobally } from '../../GlobalExports';
+import { CategoryFacetTemplates } from './CategoryFacetTemplates';
+import { CategoryValueRoot } from './CategoryValueRoot';
+import { CategoryValue } from './CategoryValue';
+import { CategoryFacetQueryController } from '../../controllers/CategoryFacetQueryController';
+import { SVGDom } from '../../utils/SVGDom';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { QueryStateModel } from '../../models/QueryStateModel';
+import 'styling/_CategoryFacet';
+import { IAttributesChangedEventArg, MODEL_EVENTS } from '../../models/Model';
+import { Utils } from '../../utils/Utils';
+import _ = require('underscore');
+
+export interface CategoryFacetOptions {
+  field: IFieldOption;
+  title?: string;
+  id: string;
+}
+
+export class CategoryFacet extends Component {
+  public categoryFacetQueryController: CategoryFacetQueryController;
+  public listenToQueryStateChange = true;
+
+  static doExport = () => {
+    exportGlobally({
+      CategoryFacet
+    });
+  };
+
+  static ID = 'CategoryFacet';
+
+  static options: CategoryFacetOptions = {
+    field: ComponentOptions.buildFieldOption({ required: true }),
+    title: ComponentOptions.buildLocalizedStringOption({
+      defaultValue: l('NoTitle')
+    }),
+    id: ComponentOptions.buildStringOption({ postProcessing: (value, options: CategoryFacetOptions) => value || (options.field as string) })
+  };
+
+  private categoryValueRoot: CategoryValueRoot;
+  private categoryFacetTemplates: CategoryFacetTemplates;
+  private facetHeader: Dom;
+  private waitElement: Dom;
+  public queryStateAttribute: string;
+
+  public categoryValueRootModule = CategoryValueRoot;
```

Comment:
What's the reason for this pattern ?

Reply:
I need to mock CategoryValueRoot for testing.